src/policy.rs

```diff
@@ -58,11 +59,13 @@ impl Policy for DefaultPolicy {
         // if the node is already quarantined
         if let Some(since) = node.logs().quarantined() {
             let duration = since.elapsed().unwrap();
+            let quarantine_duration =
+                DEFAULT_QUARANTINE_DURATION.mul(node.record().lifetime_strikes);
```

Comment:
we need to have some form of boundary though. something like `max(quarantine_duration, MAX_QUARANTINE_DURATION)`. so we don't keep quarantined object for ever.